From d02cc55e482f803c9e36313c68786a13b5b2c711 Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Tue, 15 Nov 2022 13:39:26 +0100
Subject: [PATCH] CMakeLists: Fix Cross Compile

When Cross compiling, the CMAKE_INSTALL_PREFIX (i.e. target) might be
different from the CMAKE_PREFIX_PATH (i.e. host). In order, to
properly cross-compile, use the CMAKE_PREFIX_PATH instead of
CMAKE_INSTALL_PREFIX.

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99c9e2d..7cf3b29 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@ else()
   set(PYTHON_INSTALL_DIR "lib/python${PYTHON_MAJOR_MINOR}/site-packages")
   set(SHELL_EXT "sh")
 endif()
-set(AMENT_PACKAGE_DIR "${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_DIR}/ament_package")
+set(AMENT_PACKAGE_DIR "${CMAKE_PREFIX_PATH}/${PYTHON_INSTALL_DIR}/ament_package")
 if(NOT EXISTS "${AMENT_PACKAGE_DIR}")
   # Check for an .egg-link file and use the listed directory if it exists
   get_filename_component(AMENT_PACKAGE_EGG_LINK "${AMENT_PACKAGE_DIR}" DIRECTORY)
-- 
2.37.4

